pkgs/applications/misc/bk/default.nix

```
@@ -0,0 +1,28 @@
+{ fetchCrate
+, installShellFiles
+, lib
+, pkg-config
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "bk";
+  version = "0.6.0";
+
+  src = fetchCrate {
+    inherit pname version;
+    hash = "sha256-rSMvx/zUZqRRgj48TVVG7RwQT8e70m0kertRJysDY4Y=";
+  };
+
+  cargoHash = "sha256-pE5loMwNMdHL3GODiw3kVVHj374hf3+vIDEYTqvx5WI=";
+
+  nativeBuildInputs = [ installShellFiles pkg-config ];
+
+  meta = with lib; {
+    homepage = "https://github.com/aeosynth/bk";
+    description = "a terminal epud reader written in rust";
+    license = licenses.mit;
+    maintainers = with maintainers; [ VuiMuich ];
+    platforms = lib.platforms.linux;
```

Meta mainProgram should be explicitly set (:

```suggestion
    platforms = platforms.linux;
    mainProgram = "bk";
```

---

Out of curiosity, is this a recent requirement? In previous reviews of my PRs this was never requested so far.

---

The assumption that pname and mainProgram is the same is deprecated. This is however not super well known yet and poorly documented. You can find the warning under meta.nix on line 149 (: